packages/next/next-server/server/next-server.ts

```diff
@@ -324,10 +324,40 @@ export default class Server {
 
       let defaultLocale = i18n.defaultLocale
       let detectedLocale = detectLocaleCookie(req, i18n.locales)
+
+      // Parse the `accept-language` header
+      const regex = /((([a-zA-Z]+(-[a-zA-Z0-9]+){0,2})|\*)(;q=[0-1](\.[0-9]+)?)?)*/g
```

Comment:
More tests on this would be nice. Better yet if there's already something OSS that gives you the full list we should just use that.

The regex seems to be a little too restrictive compared to the header spec expectation.

Reply:
Agreed with @longlho

Reply:
@timneutkens There are already several tests in i18n-support that test the detection from the accept-language header and they all seem to pass. How would you like me to test the regex? Is this something that should be included in the NextJS tests or would you want me to test it on my own (which is not ideal)?

Reply:
At 1st glance seems like regex doesn't fully match `es-419-nu-hebr-x-private;q=1` for example

Reply:
I created the regex based on: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html. I may have made a mistake somewhere, but for the moment, I never found an example where the regex was handling the header incorrectly. I think the best way to make sure it works is by trying to find example where it doesn't work (and if we can't then it's fine).

Reply:
hmm you prob would have to implement the full [RFC4647](https://tools.ietf.org/html/rfc4647#section-1) (sans the `lookup` section). RFC2616 has been deprecated since 2014.

Reply:
@longlho That may be me not understanding the specification correctly, but to me `es-419-nu-hebr-x-private` isn't a `language-range`? According to the specification `language-range  = ( ( 1*8ALPHA *( "-" 1*8ALPHA ) ) | "*" )`

Is that correct or am I wrong?

Reply:
The regex in the rfc you linked has been deprecated since 2014.

Reply:
Ok, so that's why it got your example wrong. What is the updated specification ?